libcudacxx/include/cuda/std/detail/libcxx/include/__utility/pair.h

```diff
@@ -69,707 +70,704 @@
+      : __base(_CUDA_VSTD::forward<_U1>(__u1), _CUDA_VSTD::forward<_U2>(__u2))
+  {}
+
+  template <class _U1          = _T1,
+            class _U2          = _T2,
+            class _Constraints = typename __pair_constraints<_T1, _T2>::template __constructible<_U1, _U2>,
+            __enable_if_t<_Constraints::__implicit_constructible, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY constexpr pair(_U1&& __u1, _U2&& __u2) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T1, _U1) && _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T2, _U2))
+      : __base(_CUDA_VSTD::forward<_U1>(__u1), _CUDA_VSTD::forward<_U2>(__u2))
+  {}
+
+  template <class... _Args1, class... _Args2>
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX17
+  pair(piecewise_construct_t __pc, tuple<_Args1...> __first_args, tuple<_Args2...> __second_args) noexcept(
+    (is_nothrow_constructible<first_type, _Args1...>::value && is_nothrow_constructible<second_type, _Args2...>::value))
+      : __base(__pc,
+               __first_args,
+               __second_args,
+               __make_tuple_indices_t<sizeof...(_Args1)>(),
+               __make_tuple_indices_t<sizeof...(_Args2)>())
+  {}
+
+  // copy constructors
+  pair(pair const&) = default;
+  pair(pair&&)      = default;
+
+  template <class _U1 = _T1,
+            class _U2 = _T2,
+            class _Constraints = typename __pair_constraints<_T1, _T2>::template __constructible<const _U1&, const _U2&>,
+            __enable_if_t<_Constraints::__explicit_constructible, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY explicit _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 pair(const pair<_U1, _U2>& __p) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T1, const _U1&)
+    && _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T2, const _U2&))
+      : __base(__p.first, __p.second)
+  {}
+
+  template <class _U1 = _T1,
+            class _U2 = _T2,
+            class _Constraints = typename __pair_constraints<_T1, _T2>::template __constructible<const _U1&, const _U2&>,
+            __enable_if_t<_Constraints::__implicit_constructible, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 pair(const pair<_U1, _U2>& __p) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T1, const _U1&)
+    && _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T2, const _U2&))
+      : __base(__p.first, __p.second)
+  {}
+
+  // move constructors
+  template <class _U1          = _T1,
+            class _U2          = _T2,
+            class _Constraints = typename __pair_constraints<_T1, _T2>::template __constructible<_U1, _U2>,
+            __enable_if_t<_Constraints::__explicit_constructible, int> = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY explicit _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 pair(pair<_U1, _U2>&& __p) noexcept(
+    _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T1, _U1) && _LIBCUDACXX_TRAIT(is_nothrow_constructible, _T2, _U2))
```

Comment:
In this constructor and related, shouldn't the test be for constructibility from `_U1&&` and `_U2&&`, not just `_U1` and `_U2`? You do add the `&&` in the test for assignment operators below, but not here, I don't see why there would be a difference, and the one on assignments feels more correct.